polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -704,10 +748,12 @@ async fn advertise_collation<Context>(
 			}
 		}
 
-		let should_advertise =
-			per_relay_parent
-				.validator_group
-				.should_advertise_to(candidate_hash, peer_ids, &peer);
+		let should_advertise = per_relay_parent
+			.validator_group
+			.get(&core_index)
+			.map_or(ShouldAdvertiseTo::NotAuthority, |group| {
```

Comment:
`NotAuthority` seems misleading. In fact this happening suggests a bug in the code right? Maybe we should just bail?

Reply:
Yeah, I reused an error which is not actually relevant.